src/python/pants/backend/tools/taplo/subsystem.py

```diff
@@ -2,6 +2,8 @@
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 from __future__ import annotations
 
+from pathlib import Path
```

Comment:
We really need a local lint about this. We prefer `PurePath` or `PureUnixPath` so that we don't accidentally start reading from disk.

```suggestion
from pathlib import PureUnixPath
```